### PR TITLE
Use a 32 bit seed for new game seeding in HQ mode.

### PR DESCRIPTION
**Pull request by tertu-m**
Also adjust some comments.

## Description
Use timer 1 and timer 2 together for seeding during new games if `HQ_RANDOM` is enabled.

## **Discord contact info**
tertu

### CODE REVIEW

**Comment by LOuroboros** [comment]
You might want to avoid using terms like "TM1" and "TM2" for stuff that is unrelated to technical machines, since that would taint searches involving TMs 10 to 29. I could see that being a small pita for users.

File: src/main.c [hunk 1]
```diff
@@ -215,15 +215,35 @@ void SetMainCallback2(MainCallback callback)
 
 void StartTimer1(void)
 {
+    #if HQ_RANDOM == TRUE
+        REG_TM2CNT_L = 0;
+        // Set timer 2 to count timer 1 overflows.
+        REG_TM2CNT_H = 0x84;
```
Comment (mrgriffin) [inline src/main.c:221]:
```suggestion
        REG_TM2CNT_H = TIMER_ENABLE | TIMER_COUNTUP;
```

**Review by mrgriffin — CHANGES_REQUESTED** [review]
How doable do you think it would be to use `if (HQ_RANDOM)` rather than `#if HQ_RANDOM == TRUE`?

File: src/main.c (hunk 1 above)
Comment (tertu-m) [reply src/main.c]: Addressed both of these.

**Review by mrgriffin — APPROVED** [review]

**Pull request merged**